src/main/java/org/kohsuke/groovy/sandbox/SandboxTransformer.java

```diff
@@ -704,7 +704,7 @@ public void visitExpressionStatement(ExpressionStatement es) {
                     if (mightBePositionalArgumentConstructor((VariableExpression) leftExpression)) {
                         CastExpression ce = new CastExpression(leftExpression.getType(), de.getRightExpression());
                         ce.setCoerce(true);
-                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), transform(ce))));
+                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), ce)));
```

Comment:
What I think are appropriate tests have been added to https://github.com/jenkinsci/script-security-plugin/pull/139 and https://github.com/jenkinsci/workflow-cps-plugin/pull/164